code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -175,38 +175,91 @@
+		to_chat(src,"<span class='warning'>You can't communicate while unable to move your hands to your head!</span>")
+		return FALSE
+	if(last_special > world.time)
+		to_chat(src,"<span class='notice'>Your mind requires rest!</span>")
+		return FALSE
+
+	last_special = world.time + 100
+
+	visible_message("<span class='notice'>[src] touches their fingers to their temple.</span>")
 
 	var/list/targets = list()
-	var/target = null
+	for(var/mob/living/M in view(client.view, client.eye))
+		targets += M
+	var/mob/living/target = null
 	var/text = null
 
-	targets += getmobs() //Fill list, prompt user with list
 	target = input("Select a creature!", "Speak to creature", null, null) as null|anything in targets
 
-	if(!target) return
+	if(!target)
+		return
 
 	text = input("What would you like to say?", "Speak to creature", null, null)
 
 	text = sanitize(text)
 
-	if(!text) return
+	if(!text)
+		return
 
-	var/mob/M = targets[target]
+	if(istype(target, /mob/abstract/observer) || target.stat == DEAD)
+		to_chat(src,"<span class='warning'>Not even a [src.species.name] can speak to the dead.</span>")
+		return
 
-	if(istype(M, /mob/abstract/observer) || M.stat == DEAD)
-		src << "Not even a [src.species.name] can speak to the dead."
+	if (target.isSynthetic())
+		to_chat(src,"<span class='warning'>This can only be used on living organisms.</span>")
 		return
 
-	log_say("[key_name(src)] communed to [key_name(M)]: [text]",ckey=key_name(src))
 
-	M << "<span class='notice'>Like lead slabs crashing into the ocean, alien thoughts drop into your mind: [text]</span>"
-	if(istype(M,/mob/living/carbon/human))
-		var/mob/living/carbon/human/H = M
-		if(H.species.name == src.species.name)
-			return
-		H << "<span class='warning'>Your nose begins to bleed...</span>"
-		H.drip(1)
+	if (target.is_diona())
+		to_chat(src,"<span class='alium'>The creature's mind is not solid enough and slips through like sand.</span>")
+		return
+
+	if(!(target in view(client.view, client.eye)))
+		to_chat(src,"<span class='warning'>[target] is too far for your mind to grasp!</span>")
+		return
+
+	log_say("[key_name(src)] communed to [key_name(target)]: [text]",ckey=key_name(src))
+
+	for (var/mob/M in player_list)
+		if (istype(M, /mob/abstract/new_player))
+			continue
+		else if(M.stat == DEAD &&  M.client.prefs.toggles & CHAT_GHOSTEARS)
+			to_chat(M,"<span class='notice'>[src] telepathically says to [target], [text]")
+
+	var/mob/living/carbon/human/H = target
+	if (/mob/living/carbon/human/proc/commune in target.verbs)
+		to_chat(H,"<span class='psychic'>You instinctively sense [src] sending their thoughts into your mind, hearing:</span> [text]")
+	else if(prob(25) && (target.mind && target.mind.assigned_role=="Chaplain"))
+		to_chat(H,"<span class='changeling'>You sense [src]'s thoughts enter your mind, whispering quietly:</span> [text]")
+	else
+		to_chat(H,"<span class='alium'>You feel pressure behind your eyes as alien thoughts enter your mind:</span> [text]")
+		if(H)
```

Comment:
This is unnecessary. `if (!target)` is checked against on line 205. So `H` will always be `true` after that. Unless you mean to do `if (istype(H))` which you absoloutely should do.